core/state.go

```diff
@@ -390,3 +399,18 @@ func (s *State) updateDeclaredClasses(declaredClasses []DeclaredV1Class) error {
 
 	return classesCloser()
 }
+
+// ContractIsDeployedAt returns if contract at given addr was deployed at blockNumber
+func (s *State) ContractIsDeployedAt(addr *felt.Felt, blockNumber uint64) (bool, error) {
```

Comment:
Shouldn't the function name be more like `ContractIsDeployedAtOrBefore`?

Reply:
Probably yes, I just wanted to keep the naming consistent with the other functions in the same interface.

How does `ContractIsAlreadyDeployedAt` sound?

Reply:
Yeah, sound much better to me

Reply:
Done